processing/src/main/java/io/druid/query/groupby/GroupByQueryQueryToolChest.java

```diff
@@ -351,6 +351,11 @@ public String apply(DimensionSpec input)
       private final List<AggregatorFactory> aggs = query.getAggregatorSpecs();
       private final List<DimensionSpec> dims = query.getDimensions();
 
+      @Override
+      public boolean isCacheable(GroupByQuery query, boolean willMergeRunners)
+      {
+        return willMergeRunners || !strategySelector.useStrategyV2(query);
```

Comment:
IMO, this should be part of the strategy, since that's the point of the strategies. So the code could be:

```java
return strategySelector.strategize(query).isCacheable(willMergeRunners);
```

Along with an "isCacheable" method added to the strategies.